### ISSUE TITLE
Using slayer.neuron.alif.Neuron module without a spike function

### ISSUE THREAD

**Issue by FranziSchwaiger** [lava-nc/lava-dl#155]
Objective of issue:

**Lava DL version:**

- [ ] **0.3.0** (feature release)
- [ ] **0.2.1** (bug fixes)
- [x] **0.2.0** (current version)
- [ ] **0.1.2**

**Lava version:**

- [ ] **0.4.0** (feature release)
- [ ] **0.3.1** (bug fixes)
- [x] **0.3.0** (current version)
- [ ] **0.2.0**
- [ ] **0.1.2**

**I'm submitting a ...** 

- [ ] bug report
- [ ] feature request
- [ ] documentation request

**Current behavior:**

- 

**Expected behavior:**

- 

**Steps to reproduce:**

- 

**Related code:**

```
insert short code snippets here
```

**Other information:**

Hi,

I wonder whether there is already a workaround to use a convolution block with ALIF neurons but instead of getting spikes as output, the internal voltage over time is returned. I have tried to re-implement the slayer.neuron.alif.Neuron module without calling "self.spike(voltage, threshold, refractory)" in the forward function, but then I get issues regarding non-contiguous gradients:

![image](https://user-images.githubusercontent.com/69162944/212371162-abd2fe85-4496-4efb-a03f-c3f17b92645e.png)

Thanks a lot!
Franziska

```
insert the output from lava debug here
```

**Comment by bamsumit** [comment]
You should be able to capture the neuron dynamics using
```python
alif = slayer.neuron.alif.Neuron(...)
current, voltage, threshold, refractory = alif.dynamics(input)  # This will give you dynamics
spike = alif.spike(voltage, threshold, refractory)
# these will give you spikes directly
spike = alif(input)
```

Regarding the contiguous assert error, can you provide a minimal script for us to look at?

**Comment by FranziSchwaiger** [comment]
Thanks for the quick response!

Here is my minimal script:

```
"""This is just a reimplementation of slayer Neuron class without the spike function"""   
class NonSpikingNeuron(slayer.neuron.alif.Neuron):
    def __init__(
        self, threshold, threshold_step,
        current_decay, voltage_decay, threshold_decay, refractory_decay,
        tau_grad=1, scale_grad=1, scale=1 << 6,
        norm=None, dropout=None,
        shared_param=True, persistent_state=False, requires_grad=False,
        graded_spike=False
    ):
        super(NonSpikingNeuron, self).__init__(
            threshold, threshold_step,
            current_decay, voltage_decay, threshold_decay, refractory_decay,
            tau_grad, scale_grad, scale,
            norm, dropout,
            shared_param, persistent_state, requires_grad,
            graded_spike
        )
    
    def forward(self, input):

        _, voltage, threshold, refractory = self.dynamics(input)
        
        # return self.spike(voltage, threshold, refractory)
        return voltage

class conv1x1_voltage(nn.Module):
    
    def __init__(self, neuron_cfg, in_planes: int, out_planes: int, stride: int = 1, groups: int = 1, dilation: int = 1, weight_scale: float = 1.):
        super().__init__()
        
        self.synapse = slayer.synapse.Conv(
                in_features=in_planes,
                out_features=out_planes,
                kernel_size=1,
                stride=stride,
                groups=groups,
                dilation=dilation,
                weight_scale=weight_scale
        )
        
        neuron_cfg["threshold"] = 1000 # set to large threshold as the neurons should never spike
        neuron_cfg["voltage_decay"] = 0.0 # we want non-leaky behavior  
        neuron_cfg["threshold_decay"] = 0.0
        neuron_cfg["requires_grad"] = False
        
        self.neuron = NonSpikingNeuron(**neuron_cfg)
    
    def forward(self, X):
        
        X = self.synapse(X)
        X = self.neuron(X)
        return X
```

When I uncomment ```return self.spike(voltage, threshold, refractory)```, everything works fine, but without the spiking, I get this error:

**Comment by FranziSchwaiger** [comment]
Quick update: I solved the issue by using ```slayer.block.alif.Conv```:

```
class NonSpikingConvBlock(slayer.block.alif.Conv):
    
    def __init__(self, neuron_params: dict, in_planes: int, out_planes: int, stride: int = 1, groups: int = 1,  padding: int = 1,
        dilation: int = 1, weight_scale: float = 1.):
        
        super(NonSpikingConvBlock, self).__init__(neuron_params=neuron_params,
        in_features=in_planes,
        out_features=out_planes,
        kernel_size=1,
        stride=stride,
        padding=padding,
        groups=groups,
        dilation=dilation,
        weight_scale=weight_scale)
        
        self.neuron = NonSpikingNeuron(**neuron_params)
```
